packages/regl-worldview/src/Worldview.js

```diff
@@ -149,14 +154,20 @@ export class WorldviewBase extends React.Component<BaseProps, State> {
   }
 
   _onClick = (e: MouseEvent) => {
-    this._onMouseInteraction(e, "onClick");
+    if (!this._disableOnClick) {
+      this._onMouseInteraction(e, "onClick");
+    }
+    this._disableOnClick = false;
   };
 
   _onDoubleClick = (e: MouseEvent) => {
+    this._disableOnClick = false;
     this._onMouseInteraction(e, "onDoubleClick");
   };
 
   _onMouseDown = (e: MouseEvent) => {
+    this._disableOnClick = false;
+    this.setState({ dragStartPos: { x: e.clientX, y: e.clientY } });
```

Comment:
instead of using state for this what about just setting it as an instance variable on the component? Will remove a render & draw call that I don't think we need.

Reply:
Good idea! Will update